attribute-service/src/integrationTest/java/org/hypertrace/core/attribute/service/AttributeServiceTest.java

```diff
@@ -126,6 +128,40 @@ private void testCreateInvalid(boolean useRequestHeaders) {
     }
   }
 
+  @Test
+  public void testCheckValidAttributeMetadata() {
+    AttributeMetadata expectedAttributeMetadata =
+        AttributeMetadata.newBuilder()
+            .setFqn("name-1")
+            .setValueKind(AttributeKind.TYPE_STRING)
+            .setKey("key-1")
+            .setDisplayName("displayname-1")
+            .setScope(AttributeScope.EVENT)
+            .setMaterialized(false)
+            .setUnit("unit-1")
+            .setType(AttributeType.ATTRIBUTE)
+            .addAllLabels(List.of("label-1", "label-2"))
+            .addAllSupportedAggregations(List.of(AggregateFunction.SUM, AggregateFunction.AVG))
+            .setOnlyAggregationsAllowed(true)
+            .addSources(AttributeSource.EDS)
+            .setId("EVENT.key-1")
+            .setGroupable(true)
+            .setDefinition(AttributeDefinition.newBuilder().setSourcePath("sourcepath-1"))
+            .setScopeString(AttributeScope.EVENT.name())
+            .setInternal(true)
+            .build();
+
+    AttributeCreateRequest request =
+        AttributeCreateRequest.newBuilder().addAttributes(expectedAttributeMetadata).build();
+    client.create(requestHeaders, request);
+
+    List<AttributeMetadata> attributeMetadataList =
+        Streams.stream(
+                client.findAttributes(requestHeaders, AttributeMetadataFilter.getDefaultInstance()))
+            .collect(Collectors.toList());
+    assertEquals(expectedAttributeMetadata, attributeMetadataList.get(0));
```

Comment:
nit: if the following works
```suggestion
    assertEquals(List.of(expectedAttributeMetadata), attributeMetadataList);
```

else, please assert the size of the `attributeMetadataList` as well